addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+                if (cfg[1].equals("softreset") && value instanceof Boolean && ((Boolean) value) == true) {
+                    try {
+                        http.doGet(moduleIpAddress + "/settings?restart=true");
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    }
+                    value = false;
+                } else if (cfg[1].equals("removewifi") && value instanceof Boolean && ((Boolean) value) == true) {
+                    try {
+                        http.doGet(moduleIpAddress + "/settings?restore=true");
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    }
+                    value = false;
+                } else if (cfg[1].equals("sendConfig") && value instanceof Boolean && ((Boolean) value) == true) {
+                    try {
+                        String response = updateKonnectedModule();
+                        logger.trace("The response from the konnected module with thingID {} was {}",
+                                getThing().getUID().toString(), response);
+                        if (response == null) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Unable to communicate with Konnected Module confirm settings.");
+                        } else {
+                            updateStatus(ThingStatus.ONLINE);
+                        }
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    }
+                    value = false;
+                }
+            }
+        }
+
+        super.handleConfigurationUpdate(configurationParameters);
+        try {
+            String response = updateKonnectedModule();
+            logger.trace("The response from the konnected module with thingID {} was {}",
+                    getThing().getUID().toString(), response);
+            if (response == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to communicate with Konnected Module confirm settings.");
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            logger.trace("There was an IOEcption Error thrown during HandleConfigurationUpdate()");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            try {
+                checkConfigruation();
```

Comment:
This could safely be moved out of the scheduled job.

Reply:
moved the configuration checking out of the scheduled job